docs/architecture.md

```diff
@@ -1,8 +1,94 @@
+
+In the architecture chart above, the two CDN configurations are referred to as "outer CDN" (receives incoming requests first, serves a custom domain name, adds custom caching rules) and "inner CDN" (provides basic caching, routes requests to the various backends used by Project Helix)
+
+The CDN configuration is managed as a VCL-based Fastly service configuration (although experimental implementations based on Cloudflare Workers and Fastly Compute@Edge exist).
+
+Based on the URL, the Fastly Service performs an edge dictionary lookup to determine the appropriate Content Bus ID (a unique identifier of a content source used while publishing) and the type of request to handle.
+
+- Media Requests (images, videos, etc.) are served directly from the Media Bus using content-addressable storage, where the content hash forms the significant part of the URL
+- Content Requests (HTML and JSON) are served using the Helix Pipeline Service, see the [PIPE](#pipe) section for details
+- Static Content is served from the Content Bus directly, or if unavailable, from the Code Bus storage location.
+
+At the Inner CDN level, no caching is applied, so that all requests accurately represent the latest state of published content.
+
+### PIPE
+
+The main delivery functionality is provided by the [`helix-pipeline-service`](https://github.com/adobe/helix-pipeline-service), which uses the [`helix-pipeline`](https://github.com/adobe/helix-pipeline) framework to render HTML, and renders filtered JSON directly (after applying some filters).
+
+The Pipeline Service pulls configuration from the Code Bus and the published Content from the Content Bus.
+
+### AUTHOR
+
+Authoring in Helix 3 is handled through the interplay of three components:
+
+1. The [Helix Bot](https://github.com/adobe/helix-bot) watches a GitHub repository for code changes and invokes Helix Admin
+2. The [Helix Sidekick](https://github.com/adobe/helix-sidekick) allows authors to trigger publishing actions explicitly and invokes Helix Admin
+3. [Helix Admin](https://github.com/adobe/helix-admin) does all the work, with the help of [Helix Content Proxy](https://github.com/adobe/helix-content-proxy), which renders content from Sharepoint and Google Drive by invoking [Helix Data Embed](https://github.com/adobe/helix-data-embed), [Helix Word Markdown Adapter](https://github.com/adobe/helix-word2md) or [Helix Google Docs Markdown Adapter](https://github.com/adobe/helix-gdoc2md)
+
+The [Helix Admin REST API](https://opensource.adobe.com/helix-home/admin/) is fully documented and covers the main functionality:
+
+- Preview: generates a Markdown document from the original content source, places it in the `preview` partition of the content bus and returns the preview URL
+- Publish: copies a Markdown document from the `preview` partition into the `live` partition of the content bus
```

Comment:
That step also invokes the indexer for that document, should that be mentioned as well in this short overview, @tripodsan ?

Reply:
publish additionally does:
- purges the outer cdn
- triggers the indexer for that document